packages/blade/src/components/Dropdown/__tests__/Dropdown.native.test.tsx

```diff
@@ -171,4 +171,99 @@ describe('<Dropdown />', () => {
     expect(getByTestId('action-list-item-test')).toBeTruthy();
     expect(getByTestId('action-list-footer-test')).toBeTruthy();
   });
+
+  it('should handle controlled props with single select', () => {
+    const ControlledDropdown = (): JSX.Element => {
+      const [currentSelection, setCurrentSelection] = React.useState<undefined | string>();
+
+      return (
+        <>
+          <Button onClick={() => setCurrentSelection('bangalore')} testID="select-blr-button">
+            Select Bangalore
+          </Button>
+          <Dropdown selectionType="single">
+            <SelectInput
+              label="Select City"
+              value={currentSelection}
+              onChange={(args) => {
+                setCurrentSelection(args.values[0]);
+              }}
+            />
+            <DropdownOverlay>
+              <ActionList>
+                <ActionListItem title="Bangalore" value="bangalore" />
+                <ActionListItem title="Pune" value="pune" testID="pune-option" />
+                <ActionListItem title="Chennai" value="chennai" />
+              </ActionList>
+            </DropdownOverlay>
+          </Dropdown>
+        </>
+      );
+    };
+
+    const { getByTestId, getByRole } = renderWithTheme(<ControlledDropdown />);
+
+    const selectInput = getByRole('combobox');
+    expect(selectInput).toHaveTextContent('Select Option');
+    fireEvent.press(getByTestId('select-blr-button'));
```

Comment:
No other way to get the button? Other than using test id i.e.

Reply:
FYI, `userEvent` is a better api to write these tests. 
This blog really does a good job to explain the difference very well
https://blog.mimacom.com/react-testing-library-fireevent-vs-userevent/

Reply:
> No other way to get the button? Other than using test id i.e.

Not in react native unfortunately. The API is very limited (we can't use `name` property of getByRole in react native. Also the reason why we added `testID` to our components recently)

> FYI, userEvent is a better api to write these tests.

It is! We use it in web tests, on native its not supported.

Reply:
Doesn't `getByText`, `getByLabelText` work 🤔 used them on alert native tests

Reply:
It works when the node is just `<Text>Some Text</Text>` but doesn't work when you want to select outer element like `<Button><Text>Some Text</Text></Button>`

Reply:
Weird, I remember using same for testing the CTAs on alert

Reply:
ops yes righttt. `getByText` is working! It was some other visibility error earlier. Changed it here to `getByText`. In some scenarios where we have to select outer ActionList component or select Dropdown Overlay (which doesn't have any text inside it), I have still kept it `getByTestId`. But for ActionListItem and Button it's working fine now.